plenum/server/node.py

```diff
@@ -2171,10 +2136,13 @@ def processOrdered(self, ordered: Ordered):
 
         logger.trace("{} got ordered requests from master replica"
                      .format(self))
-        requests = [self.requests[request_id].finalised
-                    for request_id in ordered.reqIdr
-                    if request_id in self.requests and
-                    self.requests[request_id].finalised]
+
+        # TODO all reqKeys have to be in self.requests, if not we should
+        # review and verify all data come from replicas
+        requests = [
+            self.requests[reqKey].finalised for reqKey in ordered.reqIdr
```

Comment:
It is not expected as long as requests is popped out from self.requests after it has been executed and cleaned by all replica. Only in case of some bug.